packages/blade/src/components/Box/BaseBox/types/props.types.ts

```diff
@@ -129,22 +138,19 @@ type BoxProps = Partial<
     LayoutProps &
     FlexboxProps &
     PositionProps &
-    GridProps & { children?: React.ReactNode | React.ReactNode[] }
+    GridProps &
+    BoxVisualProps & { children?: React.ReactNode | React.ReactNode[] }
 >;
 
-type BaseBoxProps = BoxProps &
+// Visual props have different types for BaseBox and Box.
+// So first we Omit Visual props of Box
+// Then we append BaseBoxVisualProps and some other props for styled-components like class and id
+type BaseBoxProps = Omit<BoxProps, keyof BoxVisualProps> &
   Partial<
-    VisualProps & {
+    BaseBoxVisualProps & {
       className?: string;
       id?: string;
```

Comment:
Its on BaseBox no? we allow id and className here. styled-components itself won't really throw an error since it takes care of `className` internally but we are explicitly passing className to BaseBox in BaseText and BaseLink. It's useful for us internally.

`id` I wanted to discuss and see if we can export to public as well. Let me just wrap up these PR stacks. I have kept a todo of some props that I feel we can expose. We'll discuss these after all PRs are merged.

Reply:
Right now I am exposing id to BaseBox only. It is useful in assigning id to section so we can add scroll to anchor link to that section (like when you're building table of content). Using this in the Box's documentation itself

Reply:
BaseBox is fine but we shall not add it to consumer components otherwise people will do?
```css
#myrandonid {
 background-color: red
}
```

Reply:
That is also true 😢 

People can still do this btw but yes I get your point. It will make extending box a lot easier.
```jsx
<div id="someid">
  <Box>hello I am red background</Box/>
</div>

#someid > div {
  background-color: red;
}
```